no-multiple-empty-lines, max beginning of file bug

**What version of ESLint are you using?**
2.10.0 , sha: 8b78697f1c2f20fda8a4ec53819be76be3de2193

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
$ node --version
v5.5.0
```

**What did you do? Please include the actual source code causing the issue.**
The *no-multiple-empty-lines* rule flagged code as invalid even though it should work according to the doc. The following test case fails:

*file tests/lib/rules/no-multiple-empty-lines.js: 140*
```
{
  code: "\n\n\n\n// invalid 16\nvar a = 5;\n",
  options: [ { max: 0, maxBOF: 4 } ]
}
```
Note that this is a modified version of the invalid test case already present in the code. I'm writing a fixer for this rule and it seems no fix of invalid test case 16 is ever accepted.

**What did you expect to happen?**
I expect that the test case code provided above is seen as valid by the test runner. This should be the solution to the invalid test case 16. However it is not accepted. Moreover there doesn't seem to be a permutation which valid. It seems the rule cannot handle a `maxBOF` > `max`. Given these options the rule is simply never valid.

**What actually happened? Please include the actual, raw output from ESLint.**
The test case returns an error:

```
$ npm test
...
  1) no-multiple-empty-lines valid



// invalid 16
var a = 5;
:

      AssertionError: Should have no errors but had 1: [ { ruleId: 'no-multiple-empty-lines',
    severity: 1,
    message: 'More than 0 blank lines not allowed.',
    line: 4,
    column: 2,
    nodeType: 'Program',
    source: '' } ]
      + expected - actual

      -1
      +0

      at testValidTemplate (lib/testers/rule-tester.js:9:9813)
      at Context.<anonymous> (lib/testers/rule-tester.js:9:14513)
...
```

--- a/lib/rules/no-multiple-empty-lines.js
+++ b/lib/rules/no-multiple-empty-lines.js
@@ -52,8 +52,8 @@ module.exports = {
 
         if (context.options.length) {
             max = context.options[0].max;
-            maxEOF = context.options[0].maxEOF;
-            maxBOF = context.options[0].maxBOF;
+            maxEOF = typeof context.options[0].maxEOF !== "undefined" ? context.options[0].maxEOF : max;
+            maxBOF = typeof context.options[0].maxBOF !== "undefined" ? context.options[0].maxBOF : max;
         }
 
         var sourceCode = context.getSourceCode();
@@ -122,11 +122,10 @@ module.exports = {
 
                 // Aggregate and count blank lines
                 if (firstNonBlankLine > maxBOF) {
-                    currentLocation = firstNonBlankLine - 1;
-
                     context.report(node, 0,
                             "Too many blank lines at the beginning of file. Max of " + maxBOF + " allowed.");
                 }
+                currentLocation = firstNonBlankLine - 1;
 
                 lastLocation = currentLocation;
                 currentLocation = trimmedLines.indexOf("", currentLocation + 1);

--- a/tests/lib/rules/no-multiple-empty-lines.js
+++ b/tests/lib/rules/no-multiple-empty-lines.js
@@ -137,6 +137,14 @@ ruleTester.run("no-multiple-empty-lines", rule, {
             code: "`\n\n\n\n\n`\n// valid 15\nvar a = 5;",
             options: [ { max: 0, maxBOF: 0 } ],
             parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "\n\n\n\n// valid 16\nvar a = 5;\n",
+            options: [ { max: 0, maxBOF: 4 } ]
+        },
+        {
+            code: "// valid 17\nvar a = 5;\n\n",
+            options: [ { max: 0, maxEOF: 1 } ]
         }
     ],
 
@@ -153,12 +161,12 @@ ruleTester.run("no-multiple-empty-lines", rule, {
         },
         {
             code: "// invalid 3\nvar a = 5;\n\n\n\n",
-            errors: [ getExpectedError(2) ],
+            errors: [ getExpectedErrorEOF(2) ],
             options: [ { max: 2 } ]
         },
         {
             code: "// invalid 4\nvar a = 5;\n \n \n \n",
-            errors: [ getExpectedError(2) ],
+            errors: [ getExpectedErrorEOF(2) ],
             options: [ { max: 2 } ]
         },
         {
@@ -183,7 +191,7 @@ ruleTester.run("no-multiple-empty-lines", rule, {
         },
         {
             code: "// invalid 9\nvar a=5;\n\n\n\n\n",
-            errors: [ getExpectedError(2) ],
+            errors: [ getExpectedErrorEOF(2) ],
             options: [ { max: 2 } ]
         },
         {

